api/base.proto

```diff
@@ -149,3 +150,16 @@ message ConfigSource {
     AggregatedConfigSource ads = 3;
   }
 }
+
+// Configuration for low level socket in listeners and clusters. If the configuration is empty,
+// the default low level socket configuration will be choosed based on the platform and
+// existence of tls_context.
+message LowLevelSocket {
+  // The name of low level socket to instantiate. The name must match a supported low level socket
+  // implementation.
+  string name = 1;
```

Comment:
what are the valid names? Are you thinking that most of the time, people will have to specify the platform name or something and we can infer based on this? Like linux/darwin/windows/etc. ?

Reply:
No, it is same concept as filters / tracers, based on what you linked into the binary. You don't need to specify ones.